src/App.tsx

```diff
@@ -16,46 +17,84 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [isReversed, setReversed] = useState(false);
+  const [sortBy, setSortBy] = useState('');
+  const [goods, setGoods] = useState([...goodsFromServer]);
+
+  const reset = () => {
+    setGoods([...goodsFromServer]);
+    setReversed(false);
+    setSortBy('');
+  };
+
+  const reverse = () => {
+    setGoods([...goods].reverse());
+    setReversed(!isReversed);
+  };
+
+  const sortByLength = () => {
+    setGoods([...goods].sort((g1, g2) => g1.length - g2.length));
+    setSortBy('length');
+  };
+
+  const sortByAlphabet = () => {
+    setGoods([...goods].sort((g1, g2) => g1.localeCompare(g2)));
+    setSortBy('alphabet');
+  };
```

Comment:
lets create separate function which will accept goods array and sort option - and will return sorted array instead of sorting in this methods